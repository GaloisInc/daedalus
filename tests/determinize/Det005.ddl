-- testing LL(k) on lexicalistic rules

def Main = { AnyWS ; END }

def $lf                   = 10
def $cr                   = 13
def $simpleWS             = 0 | 9 | 12 | 32

def SimpleEOL             = { $cr; $lf } <| $lf
def EOL                   = SimpleEOL <| $cr
def Comment               = { Match "%"; Many $[! ($lf | $cr)]; EOL }
def AnyWS                 = $simpleWS <| Comment <| EOL


