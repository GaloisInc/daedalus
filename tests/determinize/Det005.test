Det005.ddl
--dump-core
--determinize
