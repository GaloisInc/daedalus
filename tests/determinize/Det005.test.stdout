module DaedalusMain where
 
 
-------------
 
-- ./Det005.ddl:7:1--7:43
byteset $simpleWS() : Bool =
  { 0 @(Word 8) } | { 9 @(Word 8) } | { 12 @(Word 8) } | { 32 @(Word 8) }
 
-- ./Det005.ddl:6:1--6:30
byteset $cr() : Bool =
  { 13 @(Word 8) }
 
-- ./Det005.ddl:5:1--5:30
byteset $lf() : Bool =
  { 10 @(Word 8) }
 
-- ./Det005.ddl:9:1--9:47
proc SimpleEOL_() : () =
  -- ./Det005.ddl:9:29--9:47
  -- DETERMINIZE 1 Fully
  do _x0 <- match1 { ... }
     case _x0 of
       13 ->
         -- ./Det005.ddl:9:31--9:40
         -- ./Det005.ddl:9:36--9:38
         match1_ $lf()
       10 ->
         -- ./Det005.ddl:9:45--9:47
         pure ()
       _ ->
         fail_sys @()
 
-- ./Det005.ddl:10:1--10:44
proc EOL_() : () =
  -- ./Det005.ddl:10:29--10:44
  -- DETERMINIZE 1
  do _x0 <- match1 { ... }
     case _x0 of
       13 ->
         -- OR RECONSTRUCTION
         try
           -- ./Det005.ddl:10:29--10:37
           -- ./Det005.ddl:9:31--9:40
           -- ./Det005.ddl:9:36--9:38
           match1_ $lf()
         orelse
           -- ./Det005.ddl:10:42--10:44
           pure ()
       10 ->
         -- ./Det005.ddl:10:29--10:37
         -- ./Det005.ddl:9:29--9:47
         -- ./Det005.ddl:9:45--9:47
         pure ()
       _ ->
         fail_sys @()
 
-- ./Det005.ddl:11:1--11:69
proc Comment_() : () =
  -- ./Det005.ddl:11:31--11:69
  do -- ./Det005.ddl:11:31--11:39
     match_ "%"
     -- ./Det005.ddl:11:42--11:69
     -- ./Det005.ddl:11:42--11:62
     Many_26()
     -- ./Det005.ddl:11:65--11:67
     EOL_()
 
-- ./Det005.ddl:12:1--12:55
proc AnyWS_() : () =
  -- ./Det005.ddl:12:29--12:55
  -- DETERMINIZE 1
  do _x0 <- match1 { ... }
     case _x0 of
       0 ->
         -- ./Det005.ddl:12:29--12:48
         -- ./Det005.ddl:12:29--12:37
         pure ()
       9 ->
         -- ./Det005.ddl:12:29--12:48
         -- ./Det005.ddl:12:29--12:37
         pure ()
       12 ->
         -- ./Det005.ddl:12:29--12:48
         -- ./Det005.ddl:12:29--12:37
         pure ()
       32 ->
         -- ./Det005.ddl:12:29--12:48
         -- ./Det005.ddl:12:29--12:37
         pure ()
       37 ->
         -- ./Det005.ddl:12:29--12:48
         -- ./Det005.ddl:12:42--12:48
         -- ./Det005.ddl:11:31--11:69
         -- ./Det005.ddl:11:42--11:69
         do -- ./Det005.ddl:11:42--11:62
            Many_26()
            -- ./Det005.ddl:11:65--11:67
            EOL_()
       13 ->
         -- OR RECONSTRUCTION
         -- ./Det005.ddl:12:53--12:55
         -- ./Det005.ddl:10:29--10:44
         try
           -- ./Det005.ddl:10:29--10:37
           -- ./Det005.ddl:9:31--9:40
           -- ./Det005.ddl:9:36--9:38
           match1_ $lf()
         orelse
           -- ./Det005.ddl:10:42--10:44
           pure ()
       10 ->
         -- ./Det005.ddl:12:53--12:55
         -- ./Det005.ddl:10:29--10:44
         -- ./Det005.ddl:10:29--10:37
         -- ./Det005.ddl:9:29--9:47
         -- ./Det005.ddl:9:45--9:47
         pure ()
       _ ->
         fail_sys @()
 
-- entry pont
-- ./Det005.ddl:3:1--3:26
proc Main() : () =
  -- ./Det005.ddl:3:14--3:26
  do -- ./Det005.ddl:3:14--3:18
     AnyWS_()
     -- ./Det005.ddl:3:22--3:26
     -- ./Det005.ddl:3:22--3:24
     matchEnd
 
proc Many_26() : () =
  do _x27 <- try
               do ManyBody_25()
                  pure True
             orelse
               pure False
     case _x27 of
       True ->
         Many_26()
       False ->
         pure ()
 
proc ManyBody_25() : () =
  -- ./Det005.ddl:11:47--11:62
  match1_ !($lf() | $cr())
